lld/MachO/Driver.cpp

```diff
@@ -1863,6 +1890,17 @@ bool link(ArrayRef<const char *> argsArr, llvm::raw_ostream &stdoutOS,
     config->installName = config->finalOutput;
   }
 
+  auto getClientName = [&]() {
+    StringRef cn = path::filename(config->finalOutput);
+    cn.consume_front("lib");
+    auto firstDot = cn.find_first_of('.');
+    cn = cn.take_front(firstDot);
+    auto firstUnderscore = cn.find_first_of('_');
+    cn = cn.take_front(firstUnderscore);
+    return cn;
```

Comment:
This is a little inefficient, but I erred on mirroring [how `ld64` does it](https://github.com/apple-oss-distributions/ld64/blob/47f477cb721755419018f7530038b272e9d0cdea/src/ld/InputFiles.cpp#L904-L919) as closely as possible. Happy to change this up to something more efficient if desired.